clean up lib functions

Maxez opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes shortly -->
- every function in /lib starts with `lib_` prefix,
- one main /lib header (`lib.h`) includes all other headers,
- varargs moved from hal/types.h to lib/stdarg.h,
- added sprintf() function,
- strtoul() and strtol() return unsigned long and long,

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
Code clean up.
[JIRA: PD-156](https://jira.phoenix-rtos.com/browse/PD-156)

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

<!--- In case of breaking change - please advice here what needs to be done in dependent projects. -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: ia32-generic.

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [x] I will merge this PR by myself when appropriate.

Maxez commented on lib/log.h:
```diff
@@ -33,4 +25,8 @@ extern int log_getEcho(void);
 
 
+#define log_info(fmt, ...)  do { if (log_getEcho()) lib_printf(fmt, ##__VA_ARGS__); } while (0)
+#define log_error(fmt, ...) do { lib_printf(CONSOLE_RED fmt CONSOLE_NORMAL, ##__VA_ARGS__); } while (0)
```
Fixed.

agkaminski commented on lib/strtoul.c:
```diff
@@ -50,13 +51,13 @@ unsigned int lib_strtoul(char *nptr, char **endptr, int base)
 	if (*nptr == '-') {
 		sign = -1;
-		++nptr;
+		nptr++;
```
Why this change? We preffer `++nptr`

Maxez replied:
I think that in most of our codebase we use postincrementation (just my observation) so I tried to keep the code more consistent. Should we prefer one over the other when it doesn't change the logic?

Maxez replied:
Reverted back to preinrcementation.

agkaminski commented on lib/format.c:
```diff
@@ -452,9 +438,8 @@ static void format_sprintf_num(void *ctx, feedfunc feed, u64 num64, u32 flags, i
-			int i;
 			// 30 bits
-			for (i = 0; i < 10; ++i) {
+			for (i = 0; i < 10; i++) {
```
We prefer ++i

Maxez replied:
And here is the answer for my question above :P I'll change it back to postincrement.

Maxez replied:
Fixed.

agkaminski commented on lib/format.c:
```diff
@@ -411,36 +400,33 @@ static void format_sprintf_num(void *ctx, feedfunc feed, u64 num64, u32 flags, i
 		if (flags & FLAG_64BIT) {
-			int i;
-			for (i = 0; i < 8; ++i) {
+			for (i = 0; i < 8; i++) {
```
We prefer ++i

Maxez replied:
Reverted back to ++i.

HBuczynski reviewed (changes requested):
Please update authors in the files where you have made changes.

Maxez commented on lib/format.c:
```diff
@@ -37,7 +36,7 @@
 #define GET_UNSIGNED(number, flags, args) \
 	do { \
-		if ((flags)&FLAG_64BIT) \
+		if ((flags) & FLAG_64BIT) \
```
We do want to use spaces around bit and operator.

HBuczynski commented on lib/ctype.c:
```diff
@@ -1,9 +1,9 @@
+ * Operating system loader
  *
- * ctype - code derived from libphoenix
+ * Character type
```
The comment that code is derived from libphoenix should be left to know its origin.
I think it will be important if the common part is shared between plo, kernel and libphoenix.

Maxez replied:
Restored libphoenix origin info.

HBuczynski commented on lib/ctype.h:
```diff
@@ -1,9 +1,9 @@
+ * Operating system loader
  *
- * ctype - code derived from libphoenix
+ * Character type
```
The comment that code is derived from libphoenix should be left to know its origin.
I think it will be important if the common part is shared between plo, kernel and libphoenix.

Maxez replied:
Restored libphoenix origin info. Same for the rest of files that were marked so.

Maxez commented on lib/lib.h:
```diff
@@ -13,25 +13,41 @@
-#define min(a, b) ((a > b) ? b : a)
+#define min(a, b) ({ \
+	__typeof__ (a) _a = (a); \
+	__typeof__ (b) _b = (b); \
```
Fixed.

Maxez commented on lib/lib.h:
```diff
@@ -13,25 +13,41 @@
+
+#define max(a, b) ({ \
+	__typeof__ (a) _a = (a); \
+	__typeof__ (b) _b = (b); \
```
Fixed.

Maxez commented:
> 
> 
> Please update authors in the files where you have made changes.

I've added myself to the authors list in files in which I modified the code (skipped files with fixed typos/comments).
